src/simple_graph.py

```
@@ -121,19 +162,41 @@ def neighbors(self, n):
             raise ValueError('Node is not contained within graph.')
         return node_list
 
+    def neighbors_by_name(self, name):
+        """Return neighbors connected to node called name."""
+        try:
+            node_list = self.node_dict[name].neighbors
+        except AttributeError:
+            raise TypeError('Must pass a node to neighbors method.')
+        except KeyError:
+            raise ValueError('Node is not contained within graph.')
+        return node_list
+
     def adjacent(self, n1, n2):
         """Return True if there is an edge connecting n1 -> n2.
 
         False if not, raises an error if either of the supplied nodes
         are not in g.
         """
         try:
-            n_adjacent = self.node_dict[n2.name].name in self.node_dict[n1.name].neighbors
+            for item in self.node_dict[n1.name].neighbors:
+                if self.node_dict[n2.name].name in item:
+                    return True
+            return False
         except KeyError:
             raise ValueError('Graph does not contain both nodes.  Use has_node method.')
         except AttributeError:
             raise TypeError('Must pass node types to adjacent method.')
-        return n_adjacent
+
+    def adjacent_by_name(self, name1, name2):
+        """Same as adjacent with node names used."""
+        try:
+            for item in self.node_dict[name1].neighbors:
+                if self.node_dict[name2].name in item:
```

In the case where node2/name2 is not in the Graph at all, it should raise an error instead of just returning False.